platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/problems/AdditionalData.java

```diff
@@ -40,4 +41,19 @@ public interface AdditionalData {
      * @since 8.6
      */
     Map<String, Object> getAsMap();
+
+    /**
+     * Returns custom object
+     *
+     * @since 8.13
+     */
+    Object get();
+
+    /**
+     * Returns custom object with provided type or null if not available.
+     *
+     * @since 8.13
+     */
+    @Nullable
+    <T> T get(Class<T> type);
 }
```

Comment:
Let's add some documentation about the expectations here.
- Problems API allows the association of arbitrary additional data with reports.
- The Tooling API provides a _view_ on the additional data based on the supplied type. If the view type is incompatible, even partially, then a runtime exception is thrown (maybe we need test coverage for that)
- Similarly to Tooling API model building, inheritance (by default) is not supported, the expectation is that plain data types are used here.